packages/shared/src/authz/policies/canQueryProject.ts

```diff
@@ -18,44 +16,45 @@ import {
   WorkspaceNoAccessError,
   WorkspaceSsoSessionInvalidError
 } from '../domain/authErrors.js'
+import { err, isOk, ok } from 'true-myth/result'
+import { AuthCheckContextLoaderKeys } from '../domain/loaders.js'
+import { LogicError } from '../domain/errors.js'
 
-export const canQueryProjectPolicyFactory =
-  (
-    loaders: Pick<
-      AuthCheckContextLoaders,
-      | 'getEnv'
-      | 'getProject'
-      | 'getProjectRole'
-      | 'getServerRole'
-      | 'getWorkspaceRole'
-      | 'getWorkspaceSsoProvider'
-      | 'getWorkspaceSsoSession'
-    >
-  ) =>
-  async ({
-    userId,
-    projectId
-  }: UserContext & ProjectContext): Promise<
-    AuthResult<
-      | typeof ProjectNotFoundError
-      | typeof ProjectNoAccessError
-      | typeof WorkspaceNoAccessError
-      | typeof WorkspaceSsoSessionInvalidError
-    >
-  > => {
-    const { FF_ADMIN_OVERRIDE_ENABLED, FF_WORKSPACES_MODULE_ENABLED } = loaders.getEnv()
+export const canQueryProjectPolicyFactory: AuthPolicyFactory<
+  | typeof AuthCheckContextLoaderKeys.getEnv
+  | typeof AuthCheckContextLoaderKeys.getProject
+  | typeof AuthCheckContextLoaderKeys.getProjectRole
+  | typeof AuthCheckContextLoaderKeys.getServerRole
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceRole
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceSsoProvider
+  | typeof AuthCheckContextLoaderKeys.getWorkspaceSsoSession,
+  UserContext & ProjectContext,
+  | typeof ProjectNotFoundError
+  | typeof ProjectNoAccessError
+  | typeof WorkspaceNoAccessError
+  | typeof WorkspaceSsoSessionInvalidError
+> =
+  (loaders) =>
+  async ({ userId, projectId }) => {
+    const env = await loaders.getEnv()
+    if (!isOk(env)) {
+      throw new LogicError('Failed to load environment variables')
```

Comment:
I was debating whether this should be a throw or a true-myth err() result. I thought that this is essentially a LogicError - something that should never happen during runtime and so it would be annoying to have to check for a `EnvVarsNotFoundError` structure everywhere.

It's still annoying that getEnv() has to have isOk() checked on it, even tho it doesn't have any kind of error cases, but true-myth does not support returning an `ok()` struct that you don't have to check `.isOk` on. And similarly I did not want to leave one of the loaders without a true-myth return